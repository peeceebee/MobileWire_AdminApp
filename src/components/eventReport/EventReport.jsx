import React from "react";
import ReactTable from "react-table";
import "react-table/react-table.css";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";

import { faTrash } from "@fortawesome/free-solid-svg-icons";
library.add(faTrash);

export const EventReport = props => {
  const getPostn = signin => {
    return props.positions.filter(
      postn => postn.positionId === signin.positionId
    );
  };
  const getEmpl = signin => {
    let empl = props.employees.filter(
      empl => empl.employeeId === signin.employeeId
    );
    empl = empl.pop();
    return `${empl.firstName} ${empl.lastName}`;
  };

  const signins = props.signins.map(signin => {
    const postn = getPostn(signin);
    const emplName = getEmpl(signin);
    return {
      name: emplName,
      position: postn.pop().title,
      barcode: signin.lastBarcode,
      employeeId: signin.employeeId,
      signInEventId: signin.signInEventId
    };
  });

  const columns = [
    {
      Header: "Name",
      accessor: "name"
    },
    {
      Header: "Barcode",
      accessor: "barcode"
    },
    {
      Header: "Position",
      accessor: "position"
    },
    {
      Header: "",
      accessor: "isLocked",
      Cell: e => (
        <button onClick={() => props.removeSignIn(e)}>
          <FontAwesomeIcon className="event-icon" icon="trash" />
        </button>
      )
    }
  ];

  return (
    <ReactTable
      data={signins}
      columns={columns}
      className="-striped -highlight"
    />
  );
};

EventReport.propTypes = {
  signins: PropTypes.arrayOf(PropTypes.object),
  employees: PropTypes.arrayOf(PropTypes.object),
  positions: PropTypes.arrayOf(PropTypes.object)
};
