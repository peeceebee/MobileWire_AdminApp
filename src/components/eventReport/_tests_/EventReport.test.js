import { EventReport } from "../EventReport";
import React from "react";
import { shallow } from "enzyme";

describe("EventReport", () => {
  let wrapper;
  const mockProps = {
    employees: [
      {
        cellPhone: "7277547976",
        employeeId: 1,
        firstName: "Peter",
        isTerminated: false,
        lastBarcode: "S01P002",
        lastName: "Baxter",
        lastPosition: 7
      },
      {
        cellPhone: "7277547976",
        employeeId: 2,
        firstName: "Max",
        isTerminated: false,
        lastBarcode: "S01P003",
        lastName: "Silver",
        lastPosition: 8
      }
    ],
    positions: [
      { positionId: 8, title: "Electrical Helper", abbreviation: "EH" },
      { positionId: 5, title: "Electrical Supervisor", abbreviation: "ES" },
      { positionId: 7, title: "Electrician", abbreviation: "ELEC" }
    ],
    signins: [
      {
        employeeId: 1,
        positionId: 5,
        signInEmployeeId: 144,
        signInEventId: 128
      },
      {
        signInEventId: 129,
        signInEmployeeId: 145,
        employeeId: 2,
        positionId: 8
      }
    ]
  };
  beforeEach(() => {
    wrapper = shallow(<EventReport {...mockProps} />);
  });
  it("Should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("getPostn", () => {
    it.skip("Should return the name of the position given a signIn", () => {
      const mockSignin = {
        employeeId: 1,
        positionId: 5,
        signInEmployeeId: 144,
        signInEventId: 128
      };
      const actual = wrapper.props('getEmpl');
    });
  });
});
