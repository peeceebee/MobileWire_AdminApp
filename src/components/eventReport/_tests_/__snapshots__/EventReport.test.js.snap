// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventReport Should match the snapshot 1`] = `ShallowWrapper {}`;
